lib/exqlite/sqlite3.ex

```diff
@@ -110,25 +110,91 @@ defmodule Exqlite.Sqlite3 do
   @spec prepare(db(), String.t()) :: {:ok, statement()} | {:error, reason()}
   def prepare(conn, sql), do: Sqlite3NIF.prepare(conn, sql)
 
-  @spec bind(db(), statement(), nil) :: :ok | {:error, reason()}
-  def bind(conn, statement, nil), do: bind(conn, statement, [])
-
-  @spec bind(db(), statement(), list()) :: :ok | {:error, reason()}
-  def bind(conn, statement, args) do
-    Sqlite3NIF.bind(conn, statement, Enum.map(args, &convert/1))
-  rescue
-    err in ErlangError ->
-      case err do
-        %{original: %{message: message, argument: argument}} ->
-          reraise Exqlite.BindError,
-                  [message: message, argument: argument],
-                  __STACKTRACE__
-
-        %{reason: message} ->
-          reraise Exqlite.BindError, [message: message], __STACKTRACE__
-      end
+  @doc """
+  Returns number of SQL parameters in a prepared statement.
+
+      iex> {:ok, conn} = Sqlite3.open(":memory:", [:readonly])
+      iex> {:ok, stmt} = Sqlite3.prepare(conn, "SELECT ?, ?")
+      iex> Sqlite3.bind_parameter_count(stmt)
+      2
+
+  """
+  @spec bind_parameter_count(statement) :: integer
+  def bind_parameter_count(stmt), do: Sqlite3NIF.bind_parameter_count(stmt)
+
+  @type bind_value ::
+          NaiveDateTime.t()
+          | DateTime.t()
+          | Date.t()
+          | Time.t()
+          | number
+          | iodata
+          | {:blob, iodata}
+          | atom
+
+  @deprecated "Use `bind/2` instead"
+  @spec bind(db, statement, [bind_value]) :: :ok
+  def bind(_conn, stmt, args), do: bind(stmt, args)
```

Comment:
I think all functions that receive both db and stmt can be updated to only require stmt since it has all the necessary info and we can use `sqlite3_db_handle(stmt)` to get the DB if needed.